spec/src/main/asciidoc/configsources.asciidoc

```diff
@@ -147,6 +147,17 @@ public class ExampleYamlConfigSourceProvider
 
 Please note that a single `ConfigSource` should be either registered directly or via a `ConfigSourceProvider`, but never both ways.
 
+=== Dynamic ConfigSource
+A dynamic ConfigSource is a `ConfigSource` where its content can be updated, such as updating a property name or value or adding or deleting config properties 
+from it. For the property lookup, the method `config.getValue()` or `config.getOptionalValue()` retrieves the up-to-date value.
+Alternatively, for the injection style, the following lookup should be used to retrieve the up-to-date value.
+[source, text]
+----
+    @Inject
+    @ConfigProperty(name="myprj.some.dynamic.timeout", defaultValue="100")
+    private javax.inject.Provider<Long> timeout; 
+----
+
```

Comment:
Maybe also mention that whether a `ConfigSource` supports this dynamic behavior or not is dependent on how it's implemented.

Need to make clear there's no guarantee as to whether any `ConfigSource` does, or does not, support this dynamic behavior and that it's a case by case basis